builtin/providers/azurerm/resource_arm_servicebus_topic.go

```diff
@@ -0,0 +1,222 @@
+				Computed: true,
+			},
+
+			"duplicate_detection_history_time_window": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"enable_batched_operations": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"enable_express": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"enable_filtering_messages_before_publishing": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"enable_partitioning": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"max_size_in_megabytes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"requires_duplicate_detection": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"support_ordering": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceArmServiceBusTopicCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusTopicsClient
+	log.Printf("[INFO] preparing arguments for Azure ARM ServiceBus Topic creation.")
+
+	name := d.Get("name").(string)
+	namespaceName := d.Get("namespace_name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	parameters := servicebus.TopicCreateOrUpdateParameters{
+		Name:       &name,
+		Location:   &location,
+		Properties: &servicebus.TopicProperties{},
+	}
+
+	if autoDeleteOnIdle := d.Get("auto_delete_on_idle").(string); autoDeleteOnIdle != "" {
+		parameters.Properties.AutoDeleteOnIdle = &autoDeleteOnIdle
+	}
+
+	if defaultTTL := d.Get("default_message_ttl").(string); defaultTTL != "" {
+		parameters.Properties.DefaultMessageTimeToLive = &defaultTTL
+	}
+
+	if duplicateWindow := d.Get("duplicate_detection_history_time_window").(string); duplicateWindow != "" {
+		parameters.Properties.DuplicateDetectionHistoryTimeWindow = &duplicateWindow
+	}
+
+	enableBatchedOps := d.Get("enable_batched_operations").(bool)
+	enableExpress := d.Get("enable_express").(bool)
+	enableFiltering := d.Get("enable_filtering_messages_before_publishing").(bool)
+	enablePartitioning := d.Get("enable_partitioning").(bool)
+	maxSize := int64(d.Get("max_size_in_megabytes").(int))
+	requiresDuplicateDetection := d.Get("requires_duplicate_detection").(bool)
+	supportOrdering := d.Get("support_ordering").(bool)
+
+	parameters.Properties.EnableBatchedOperations = &enableBatchedOps
+	parameters.Properties.EnableExpress = &enableExpress
+	parameters.Properties.FilteringMessagesBeforePublishing = &enableFiltering
+	parameters.Properties.EnablePartitioning = &enablePartitioning
+	parameters.Properties.MaxSizeInMegabytes = &maxSize
+	parameters.Properties.RequiresDuplicateDetection = &requiresDuplicateDetection
+	parameters.Properties.SupportOrdering = &supportOrdering
+
+	_, err := client.CreateOrUpdate(resGroup, namespaceName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, namespaceName, name)
```

Comment:
Is there any eventual consistency here or can we guarantee that this will be immediately available?